src/lib/statistics/column_statistics.cpp

```diff
@@ -0,0 +1,444 @@
+#include "column_statistics.hpp"
+
+#include <sstream>
+
+#include "resolve_type.hpp"
+#include "table_statistics.hpp"
+#include "type_cast.hpp"
+
+namespace opossum {
+
+template <typename ColumnDataType>
+ColumnStatistics<ColumnDataType>::ColumnStatistics(const float null_value_ratio, const float distinct_count,
+                                                   const ColumnDataType min, const ColumnDataType max)
+    : AbstractColumnStatistics(data_type_from_type<ColumnDataType>(), null_value_ratio, distinct_count),
+      _min(min),
+      _max(max) {
+  Assert(null_value_ratio >= 0.0f && null_value_ratio <= 1.0f, "NullValueRatio out of range");
+}
+
+template <typename ColumnDataType>
+ColumnDataType ColumnStatistics<ColumnDataType>::min() const {
+  return _min;
+}
+
+template <typename ColumnDataType>
+ColumnDataType ColumnStatistics<ColumnDataType>::max() const {
+  return _max;
+}
+
+template <typename ColumnDataType>
+std::shared_ptr<AbstractColumnStatistics> ColumnStatistics<ColumnDataType>::clone() const {
+  return std::make_shared<ColumnStatistics<ColumnDataType>>(null_value_ratio(), distinct_count(), _min, _max);
+}
+
+template <typename ColumnDataType>
+ColumnValueEstimate ColumnStatistics<ColumnDataType>::estimate_predicate_with_value(
+    const PredicateCondition predicate_condition, const AllTypeVariant& variant_value,
+    const std::optional<AllTypeVariant>& value2) const {
+  const auto value = type_cast<ColumnDataType>(variant_value);
+
+  switch (predicate_condition) {
+    case PredicateCondition::Equals:
+      return estimate_equals_with_value(value);
+    case PredicateCondition::NotEquals:
+      return estimate_not_equals_with_value(value);
+
+    case PredicateCondition::LessThan: {
+      // distinction between integers and floats
+      // for integers "< value" means that the new max is value <= value - 1
+      // for floats "< value" means that the new max is value <= value - ε
```

Comment:
Is this what the SQL standard tells us?

Reply:
I didn't change this. But imo this has nothing to do with SQL, it's just a mathematical observation.